### PR TITLE
Consolidate updating guides

### PR DESCRIPTION
**Pull request by ptgott**
Include a how-to guide for the fast path we want most users to move to: using cloud-hosted Teleport Enterprise and enrolling agents in automatic updates. For other use cases and edge cases, include a general Updating Reference.

### CODE REVIEW

File: docs/pages/upgrading/automatic-agent-updates.mdx [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+$ tctl inventory ls --upgrader=none
+Server ID                            Hostname      Services Version Upgrader
+------------------------------------ ------------- -------- ------- --------
+00000000-0000-0000-0000-000000000000 ip-10-1-6-130 Node     v14.4.5 none
+...
+```
+
+Note that the `inventory ls` command will also list `teleport-auth` and
+`teleport-proxy` services. These services are managed by the Teleport team, and
+updates will be performed automatically.
```
Comment (hugoShaka) [inline docs/pages/upgrading/automatic-agent-updates.mdx:55]: The auth and proxy sevrices are only managed by us for cloud customers, self-hosted AU users are also responsible for updating their control plane services.

File: docs/pages/upgrading/automatic-agent-updates.mdx [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+   ```code
+   $ HOST=00000000-0000-0000-0000-000000000000
+   $ USER=root
+   $ tsh ssh "${USER?}@${HOST?}"
+   ```
+
+1. Ensure that the Teleport repository is properly configured to use the
+   `stable/cloud` channel, and install the `teleport-ent-updater` package. This
+   must be completed on each agent you would like to enroll into automatic
+   updates:
```
Comment (hugoShaka) [inline docs/pages/upgrading/automatic-agent-updates.mdx:71]: This channel is cloud-specific but it seems this guide is not

File: docs/pages/upgrading/automatic-agent-updates.mdx (hunk 1 above)
Comment (ptgott) [reply docs/pages/upgrading/automatic-agent-updates.mdx:55]: I had intended for this to be a guide for cloud-hosted clusters, but now realize that nothing in the introduction/prerequisites mentions this. I've added a prerequisite for a cloud-hosted cluster and a link to the reference for self-hosted users.

File: docs/pages/upgrading/automatic-agent-updates.mdx [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+This guide explains how to enable automatic updates for Teleport agents on
+cloud-hosted Teleport Enterprise clusters.
+
+## How it works
+
+When automatic updates are enabled, a Teleport updater is installed alongside
+each Teleport agent. The updater communicates with the Teleport Proxy Service to
+determine when an update is available. When an update is available, the updater
+will update the Teleport agent during the next maintenance window. However, if a
+critical update is availble, the Teleport agent will be updated outside the
```
Comment (bernardjkim) [inline docs/pages/upgrading/automatic-agent-updates.mdx:19]:
```suggestion
critical update is available, the Teleport agent will be updated outside the
```

File: docs/pages/upgrading/automatic-agent-updates.mdx [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+- (!docs/pages/includes/cloud/tctl-tsh-prerequisite.mdx!)
+
+- (!docs/pages/includes/tctl.mdx!)
+
+Systemd agents enrolled into automatic updates can only install versions
+present in their package repositories. As Teleport 14 won't be published to
+`stable/v13`, those agents will require manual intervention to be upgraded to
+the next major version (adding a new APT/YUM/zypper repo for `stable/v14`).
+
+Alternatively, you can use the `stable/rolling` channel, which contains
```
Comment (bernardjkim) [inline docs/pages/upgrading/automatic-agent-updates.mdx:41]: I'm not really sure what the use case for `stable/rolling` was supposed to be, but for Teleport Cloud we instruct users to use the `stable/cloud` channel.

File: docs/pages/upgrading/automatic-agent-updates.mdx [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+   $ tsh ssh "${USER?}@${HOST?}"
+   ```
+
+1. Ensure that the Teleport repository is properly configured to use the
+   `stable/cloud` channel, and install the `teleport-ent-updater` package. This
+   must be completed on each agent you would like to enroll into automatic
+   updates:
+
+   ```code
+   $ curl https://goteleport.com/static/install.sh | bash -s <Var name="version" /> cloud
```
Comment (bernardjkim) [inline docs/pages/upgrading/automatic-agent-updates.mdx:80]: Users need to ensure that they install/upgrade to a compatible version of teleport here, but there isn't really any mention of what version to use here.

**Review by bernardjkim — APPROVED** [review]

**Comment by bernardjkim** [comment]
I noticed that the static install script does not configure the `stable/cloud` channel for cloud installations. I've created a PR to update the script https://github.com/gravitational/next/pull/2488. We'll probably want to merge that first.

File: docs/pages/upgrading/automatic-agent-updates.mdx (hunk 4 above)
Comment (ptgott) [reply docs/pages/upgrading/automatic-agent-updates.mdx:41]: Thanks! The two paragraphs about release channels for older Teleport versions probably shouldn't be in this page. I've removed them, and also added a subsection to the reference about choosing a release channel, based on the text in `docs/pages/includes/repo-channels.mdx`.

**Comment by bernardjkim** [comment]
We're trying to make automatic updates a requirement at this point.

Can we include more details about the Cloud updates process and use firmer language on requiring automatic updates? Something like I had in the previous PR?

```
Cloud-hosted Teleport clusters are updated on a weekly basis. Major version updates
are performed every 4 months. You can monitor and subscribe to the [Teleport Status](https://status.teleport.sh/)
page to be notified of scheduled updates.

To maintain version compatibility between your Teleport agents and the Teleport
Proxy Services, ensure that all your Teleport agents stay up to date with the Teleport
version of your Teleport Proxy Services. Failure to maintain version compatibility
for your Teleport agents may result in a degraded connection or disconnection of
those agents.
```

**Comment by ptgott** [comment]
> We're trying to make automatic updates a requirement at this point. We can't technically force automatic updates, but we should make it clear that their agents will lose compatibly if they don't keep their agents up to date with Teleport Cloud.
> 
> Can we include more details about the Cloud updates process and use firmer language on requiring automatic updates? Something like I had in the previous PR?

Added in [5c862d4](https://github.com/gravitational/teleport/pull/41160/commits/5c862d47ec1e712b88c1cee41271af383a718b2d)

**Comment by ptgott** [comment]
@r0mant Would you have a moment to take a look at this one? Thanks!

File: docs/pages/upgrading.mdx [hunk 6]
```diff
@@ -1,47 +1,22 @@
...
+understand how to upgrade components in your Teleport cluster while ensuring
+compatibility between all components.
 
-Since Teleport is a distributed system with a number of services that run on
-potentially many hosts, you should take care when upgrading the cluster to
-ensure that all components remain compatible.
+If you have a Teleport Enterprise (Cloud) account, you can [set up automatic
+Teleport agent updates](./upgrading/automatic-agent-updates.mdx) to ensure that
+the version of Teleport running on agents is always compatible with that of the
+Teleport cluster.
```
Comment (r0mant) [inline docs/pages/upgrading.mdx:16]: Can we make the language stronger here? Automatic upgrades is a requirement, not a suggestion.

File: docs/pages/upgrading/automatic-agent-updates.mdx [hunk 7]
```diff
@@ -0,0 +1,242 @@
+---
+title: Set up Automatic Agent Updates
+description: Describes how to set up automatic agent updates.
+---
+
+We strongly recommend that Teleport users set up automatic agent updates to
```
Comment (r0mant) [inline docs/pages/upgrading/automatic-agent-updates.mdx:6]: Similar to above, automatic upgrades is not a recommendation, it's a requirement.

File: docs/pages/upgrading/automatic-agent-updates.mdx [hunk 8]
```diff
@@ -0,0 +1,242 @@
+---
+title: Set up Automatic Agent Updates
+description: Describes how to set up automatic agent updates.
```
Comment (r0mant) [inline docs/pages/upgrading/automatic-agent-updates.mdx:3]: Since this is specifically for Cloud users, should we mention this in the title/description?

File: docs/config.json [hunk 9]
```diff
@@ -29,29 +29,12 @@
...
-            {
-              "title": "Self-Hosted Linux",
-              "slug": "/upgrading/self-hosted-linux/",
-              "forScopes": ["enterprise", "oss"]
-            },
-            {
-              "title": "Self-Hosted Kubernetes",
-              "slug": "/upgrading/self-hosted-kubernetes/",
-              "forScopes": ["enterprise", "oss"]
+                "title": "Set up Automatic Agent Updates",
```
Comment (r0mant) [inline docs/config.json:32]: This guide is for Cloud users it seems so I would call it out explicitly in the title.

File: docs/pages/upgrading/automatic-agent-updates.mdx [hunk 10]
```diff
@@ -0,0 +1,242 @@
...
+------------------------------------ ------------- -------- ------- --------
+00000000-0000-0000-0000-000000000000 ip-10-1-6-130 Node     v14.4.5 none
+...
+```
+
+Note that the `inventory ls` command will also list `teleport-auth` and
+`teleport-proxy` services. These services are managed by the Teleport team, and
+updates will be performed automatically.
+
+## Step 2/3. Enroll agents on Linux servers in automatic updates
```
Comment (r0mant) [inline docs/pages/upgrading/automatic-agent-updates.mdx:66]: It seems like this guide is targeting clusters with existing agents that aren't yet enrolled in auto-upgrades? We should make this clear I think. Cloud users enrolling new agents should get automatic upgrades enabled by default, is that correct?

File: docs/pages/upgrading/automatic-agent-updates.mdx [hunk 11]
```diff
@@ -0,0 +1,242 @@
...
+    2023-04-28T13:13:30Z	INFO	StatefulSet is already up-to-date, not updating.	{"controller": "statefulset", "controllerGroup": "apps", "controllerKind": "StatefulSet", "StatefulSet": {"name":"my-agent","namespace":"agent"}, "namespace": "agent", "name": "my-agent", "reconcileID": "10419f20-a4c9-45d4-a16f-406866b7fc05", "namespacedname": "agent/my-agent", "kind": "StatefulSet", "err": "no new version (current: \"v12.2.3\", next: \"v12.2.3\")"}
+    ```
+
+## Troubleshooting
+
+Teleport agents are not updated immediately when a new version of Teleport is
+released, and agent updates can lag behind the cluster by a few days.
+
+If the Teleport agent has not been automatically updating for several weeks, you
+can consult the updater logs to help troubleshoot the problem.
```
Comment (r0mant) [inline docs/pages/upgrading/automatic-agent-updates.mdx:201]: Nit: Would be helpful to show how to "consult the logs". Maybe a `journalctl` command.

File: docs/pages/upgrading/reference.mdx [hunk 12]
```diff
@@ -0,0 +1,502 @@
+---
+title: Upgrading Reference
+description: Provides detailed information on upgrading Teleport in various situations.
+tocDepth: 3
+---
+
+This guide explains how to upgrade components of a Teleport cluster in
+non-standard situations.
```
Comment (r0mant) [inline docs/pages/upgrading/reference.mdx:8]: I'm not sure why self-hosted users trying to setup auto-upgrade is a non-standard situation. Maybe rephrase this to say this is targeting self-hosted setups?

File: docs/pages/upgrading/reference.mdx [hunk 13]
```diff
@@ -0,0 +1,502 @@
...
+    ignore:
+    - paths: ["/spec/template/spec/containers/0/image"]
+      target:
+        kind: StatefulSet
+        name: teleport-agent
+        namespace: teleport
+...
+```
+
+## Self-hosted automatic updates
```
Comment (r0mant) [inline docs/pages/upgrading/reference.mdx:251]: Why is this not a separate guide, peer to Cloud automatic upgrades? Seems weird to have this hidden inside reference.

File: docs/pages/upgrading/reference.mdx [hunk 14]
```diff
@@ -0,0 +1,502 @@
...
+
+You can enable automatic agent updates in a self-hosted Teleport cluster.
+Before enabling automatic upgrades in your self-hosted Teleport cluster, you
+must enable a **version server**. Automatic upgrades in self-hosted Teleport
+clusters require at least v14.3.7 or v15.1.3.
+
+### Configuring the maintenance schedule
+
+To eanble automatic upgrades in your cluster, you must create a cluster
```
Comment (r0mant) [inline docs/pages/upgrading/reference.mdx:260]:
```suggestion
To enable automatic upgrades in your cluster, you must create a cluster
```

**Review by r0mant — COMMENTED** [review]
Left some comments.

File: docs/config.json (hunk 9 above)
Comment (ptgott) [reply docs/config.json:32]: I've made this guide for self-hosted users as well. In my original PR, the intention was to make Cloud the happy path, but I can see how that would be overly aggressive.

File: docs/pages/upgrading/automatic-agent-updates.mdx (hunk 10 above)
Comment (ptgott) [reply docs/pages/upgrading/automatic-agent-updates.mdx:66]: Edited the prerequisites. Step 2 shows the user how to identify agents that are not enrolled in automatic updates.

**Comment by ptgott** [comment]
@r0mant Friendly ping. Thanks!

**Comment by ptgott** [comment]
@r0mant Just checking if there's anything else we need to do before we can merge this simplification. Thanks!

**Review by r0mant — APPROVED** [review]
